File: java/13.java
class Solution {
    public int romanToInt(String s) {
        Map<Character, Integer> m = new HashMap<>();
        m.put('M', 1000);
        m.put('D', 500);
        m.put('C', 100);
        m.put('L', 50);
        m.put('X', 10);
        m.put('V', 5);
        m.put('I', 1);
        int result = 0;
        int n = s.length();
        for (int i = 0; i < n; i++) {
            int currentCharValue = m.get(s.charAt(i));
            if (i + 1 < n) {
                int nextCharValue = m.get(s.charAt(i + 1));
                if (currentCharValue < nextCharValue) {
                    result += nextCharValue - currentCharValue;
                    i++;
                } else {
                    result += currentCharValue;
                }
            } else {
                result += currentCharValue;
            }
        }
        return result;

    }
}

File: python/13.py
class Solution:
    def romanToInt(self, s: str) -> int:
        m = {"M": 1000, "D": 500, "C": 100, "L": 50, "X": 10, "V": 5, "I": 1}
        result = 0
        n = len(s)
        i = 0
        while i < n:
            current_char_value = m[s[i]]
            if i + 1 < n:
                next_char_value = m[s[i + 1]]
                if current_char_value < next_char_value:
                    result += next_char_value - current_char_value
                    i += 2
                else:
                    result += current_char_value
                    i += 1
            else:
                result += current_char_value
                i += 1
        return result
